sdv/tabular/base.py

```diff
@@ -504,23 +503,16 @@ def sample(self, num_rows, randomize_samples=True, batch_size=None, output_file_
 
         sampled = []
         try:
-            def _sample_function(progress_bar=None):
+            with tqdm.tqdm(total=num_rows) as progress_bar:
+                progress_bar.set_description('Sampling rows')
                 for step in range(math.ceil(num_rows / batch_size)):
                     sampled_rows = self._sample_batch(
-                        batch_size,
-                        batch_size_per_try=batch_size,
+                        batch_size=batch_size,
                         progress_bar=progress_bar,
                         output_file_path=output_file_path,
                     )
                     sampled.append(sampled_rows)
 
-                return sampled
-
-            if batch_size == num_rows:
```

Comment:
Do we still want to hide the progress bar if batch_size = num_rows? I guess it's fine to show it in that case, but it will just be one 0 -> 100 increment.

Reply:
Oh, the reason I put it is because we could get reject sampling with constraints and so the progress bar would update

Reply:
Ah okay, that's fine then.